Loop/View Controllers/SettingsTableViewController.swift

```diff
@@ -477,18 +477,28 @@ final class SettingsTableViewController: UITableViewController {
 
                 show(scheduleVC, sender: sender)
             case .suspendThreshold:
+                func presentSuspendThresholdEditor(initialValue: HKQuantity?, unit: HKUnit) {
+                    let editor = SuspendThresholdEditor(
+                        value: initialValue,
+                        unit: unit,
+                        onSave: { [dataManager, tableView] newValue in
+                            dataManager!.loopManager.settings.suspendThreshold = GlucoseThreshold(unit: unit, value: newValue.doubleValue(for: unit))
+
+                            tableView.reloadRows(at: [indexPath], with: .automatic)
+                        }
+                    )
+
+                    let hostingController = ExplicitlyDismissibleModal(rootView: editor, onDisappear: {
+                        tableView.deselectRow(at: indexPath, animated: true)
+                    })
+
+                    self.present(hostingController, animated: true)
```

Comment:
Presenting modally is a little out of place here, especially with the disclosure arrow. But I don't think this is a blocker, and the updated "save/cancel" aspect does seem a better fit for a modal.

Reply:
Yeah, modal presentation is Design's request, and I agree it's the right fit for save/cancel interactions. They're still thinking through how to update the settings cells accordingly—I think it would be neat if each setting row had its own icon (like the raised rectangle for basals, a little curve for insulin model, etc.)